File: shopbill-services/src/main/java/com/shop/bill/entity/CommonCode.java
package com.shop.bill.entity;

import javax.persistence.Entity;
import javax.persistence.Id;

import lombok.Getter;
import lombok.Setter;

/**
*
* @author Akhil Singla
*/
@Getter
@Setter
@Entity
public class CommonCode {
	@Id
	private String codeKey;
	private String parentCodeKey;
	private String codeType;
	private String messageEn;
	private String messageHi;
	private Character delYn = 'N';
}

File: shopbill-ui/src/app/shared/models/CommonCode.ts
export class CommonCode {
    codeKey: string;
	parentCodeKey: string;
	codeType: string;
	messageEn: string;
	messageHi: string;
	delYn: string;
    codes: CommonCode[]

    constructor(codeKey: string, parentCodeKey: string, codeType: string, 
        messageEn: string, messageHi: string, delYn: string, codes: CommonCode[]) {
        this.codeKey = codeKey;
        this.parentCodeKey = parentCodeKey;
        this.codeType = codeType;
        this.messageEn = messageEn;
        this.messageHi = messageHi;
        this.delYn = delYn;
        this.codes = codes;
    }
}